### ISSUE TITLE
What's a good enough loss value, in LoRA fine-tuning?

### ISSUE THREAD

**Issue by gladjoyhub** [ml-explore/mlx-examples#236]
How small should loss reach, 0.1 or 1.0?

**Comment by USMCM1A1** [comment]
Literally no idea.  My last training, 3 epochs, 1640 examples on Mistral-7b, produced a fantastic instruction-following classics model: 
![Mistral-7b-Classics-4E-164Ex-TrainingLoss](https://github.com/ml-explore/mlx-examples/assets/15972318/22b37a13-a770-48b2-a3d9-4612403c458a)

**Comment by awni** [comment]
> What's a good enough loss value, in LoRA fine-tuning? #236

I can't give you a general answer. What you should look for is improvements on the **validation** loss as that should translate to better generalization for your task. If the training loss gets too low it will start to overfit (assuming a small dataset). 

The actual loss value you that "is good enough" depends a ton on the task data and your requirements. So again I would pay more attention to changes in validation loss, which should decrease along with training loss to a point.

**Comment by USMCM1A1** [comment]
This was relatively small (1152 examples).
![training   val loss 1152 iter](https://github.com/ml-explore/mlx-examples/assets/15972318/0b95c91a-957b-4eb1-8e85-ecec10f57d15)
